server/selective_sync.go

```diff
@@ -3,41 +3,129 @@ package main
 import (
 	"math"
 
+	"github.com/mattermost/mattermost-plugin-msteams/server/store/storemodels"
 	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/pkg/errors"
 )
 
-// ChatShouldSync determines if the members of the given channel span both Mattermost and
-// MS Teams. Chats between users on the same platform are skipped if selective sync is enabled.
-// Chats with only a single member are self chats and always sync.
-func (p *Plugin) ChatShouldSync(channelID string) (bool, error) {
+func (p *Plugin) ChannelShouldSyncCreated(channelID, senderID string) (bool, error) {
```

Comment:
Why the `Created` suffix? Are the semantics unique to creation vs other events?

Reply:
With GetSyncRemoteOnly on, the created event needs to ensure the Sender is either "Connected" or "Remote". Other events don't need this check, so this was me trying to not duplicate code and  have a single place that requires slightly different functionality. I wanted to try to keep all the logic in `selective_sync.go`. I could check the connection status in the handler that makes this call.

Willing to rename, move or change login.

Reply:
`ChannelShouldSync` gets called from other places that doesn't have a `senderID`. This same code is used to determine automute. I originally wrote this as a single function and passed in "" for sender where we didn't have or need it. That to me was more confusing. I also tried doing the "connection" check outside "ChannelShouldSync", but that resulted in duplicating API calls, such as calling `ListMembers` twice. I also think this separates the code, so parts can be removed/changed, and keeps all the "selective sync" decisions in a single place.